pac4j-saml-db/src/main/java/org/pac4j/saml/dbclient/DbLoadedSamlClientConfiguration.java

```diff
@@ -0,0 +1,211 @@
+package org.pac4j.saml.dbclient;
+
+import org.opensaml.saml.common.xml.SAMLConstants;
+import org.pac4j.saml.client.SAML2ClientConfiguration;
+import org.pac4j.saml.storage.EmptyStorageFactory;
+import org.pac4j.saml.storage.SAMLMessageStorageFactory;
+
+
+/**
+ * An alternative to {@link SAML2ClientConfiguration}. Unlike {@link SAML2ClientConfiguration}, this class does not use paths to resources.
+ * Instead, it contains the data directly.
+ *
+ * Differences from {@link SAML2ClientConfiguration}:
+ * <ul>
+ * <li>The JKS keystore - contained as a byte array, not a path to a resource</li>
+ * <li>IdP metadata - contained as a string, not a path to a resource</li>
+ * <li>SP metadata - dropped completely</li>
+ * <li>Client name - added</li>
+ * </ul>
+ *
+ * TODO: It would be nice to have a common interface and maybe a common abstract parent too.
+ * 
+ * @author jkacer
+ */
+public class DbLoadedSamlClientConfiguration implements Cloneable {

```

Comment:
You should have a `SAML2DatabaseConfiguration` inheriting from `SAML2Configuration` and this latter should become an `InitializableWebObject`.
In the regular `SAML2Configuration`, the `init(context)` should do nothing while in the `SAML2DatabaseConfiguration`, the `init(context)` method should load data from database.
This `init(context)` method should be of course called itself from the `init(context)` method of the `SAML2Client`.